lib/wos_queries.rb

```diff
@@ -0,0 +1,279 @@
+          records_i = records(response_i, "#{retrieve_operation}_response".to_sym)
+          records = records.merge_records records_i
+        end
+      end
+      records
+    end
+
+    ###################################################################
+    # WoS SOAP Response Parsers
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [Hash] return data
+    def response_return(response, response_type)
+      response.body[response_type][:return]
+    end
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [Integer]
+    def query_id(response, response_type)
+      response_return(response, response_type)[:query_id].to_i
+    end
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [Integer]
+    def records_found(response, response_type)
+      response_return(response, response_type)[:records_found].to_i
+    end
+
+    # @param response [Savon::Response] a WoS SOAP response
+    # @param response_type [Symbol] a WoS SOAP response type
+    # @return [WosRecords]
+    def records(response, response_type)
+      WosRecords.new(records: response_return(response, response_type)[:records])
+    end
+
+    ###################################################################
+    # Search User Query Helpers
+
+    # Constructs a WoS name query
+    # @param name [String] a CSV name pattern: {last name}, {first_name} [{middle_name} | {middle initial}]
+    def name_query(name)
+      split_name = name.split(',')
+      last_name = split_name[0]
+      first_middle_name = split_name[1]
+      first_name = first_middle_name.split(' ')[0]
+      middle_name = first_middle_name.split(' ')[1]
+      name_query = "#{last_name} #{first_name} OR #{last_name} #{first_name[0]}"
+      name_query += " OR #{last_name} #{first_name[0]}#{middle_name[0]} OR #{last_name} #{first_name} #{middle_name[0]}" unless middle_name.blank?
+      name_query
+    end
+
+    # Search authors from these institutions
+    # @return [Array<String>] institution names
+    def institutions
+      ['Stanford University']
+    end
+
+    ###################################################################
+    # WoS Query Parameters
+
+    # @param uid [String] a WOS UID
+    # @return [Hash] citedReferences parameters
+    def cited_references_params(uid)
+      retrieve_options = [ { key: 'Hot', value: 'On' } ]
+      {
+        databaseId: @database,
+        uid: uid,
+        queryLanguage: QUERY_LANGUAGE,
+        retrieveParameters: retrieve_parameters(options: retrieve_options)
+      }
+    end
+
+    # @param uid [String] a WOS UID
+    # @return [Hash] citingArticles parameters
+    def citing_articles_params(uid)
+      {
+        databaseId: @database,
```

Comment:
Probably should have an `attr_reader :database` and use it in these methods instead of referencing the instance variable.

Reply:
Done